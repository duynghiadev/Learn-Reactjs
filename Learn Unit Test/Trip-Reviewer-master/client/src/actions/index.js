import axios from 'axios';

import {
  GET_TRIPS,
  GET_USERS,
  GET_TRIP_W_REVIEWER,
  CLEAR_TRIP_W_REVIEWER,
  GET_TRIP_W_REVIEWER_ERROR,
  USER_REGISTER,
  USER_LOGIN,
  RESET_PASSWORD,
  USER_AUTH,
  ADD_TRIP,
  UPDATE_TRIP,
  UPDATE_USER,
  DELETE_TRIP,
  GET_TRIP,
  GET_TRIP_ERROR,
  CLEAR_NEW_TRIP,
  CLEAR_TRIP,
  GET_USER_REVIEWS,
} from '../constants/action-names';

export function getTrips(limit = 10, start = 0, order = 'asc', list = []) {
  const request = axios.get(`/api/getManyTrips?limit=${limit}&skip=${start}&order=${order}`);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: GET_TRIPS,
        payload: {
          list: [...list, ...response.data],
          newTripsCount: response.data.length,
        },
      });
    });
}

export function getTripWithReviewer(id) {
  const request = axios.get(`/api/getTrip?id=${id}`);

  return dispatch =>
    request
      .then(({ data: tripData }) => {
        const trip = tripData;

        return axios.get(`/api/getReviewer?id=${trip.ownerId}`).then(({ data: reviewerData }) => {
          const response = {
            trip,
            reviewer: reviewerData,
          };

          dispatch({
            type: GET_TRIP_W_REVIEWER,
            payload: response,
          });
        });
      })
      .catch(error =>
        dispatch({
          type: GET_TRIP_W_REVIEWER_ERROR,
          payload: error,
        })
      );
}

export function getUserReviews(userId) {
  const request = axios.get(`/api/getUserReviews?user=${userId}`);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: GET_USER_REVIEWS,
        payload: response.data,
      });
    });
}

export function updateUser(data) {
  const request = axios.post('/api/userUpdate', data);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: UPDATE_USER,
        payload: response.data,
      });
    });
}

export function getTrip(id) {
  const request = axios.get(`/api/getTrip?id=${id}`);

  return dispatch =>
    request
      .then(response => {
        dispatch({
          type: GET_TRIP,
          payload: response.data,
        });
      })
      .catch(error => {
        dispatch({
          type: GET_TRIP_ERROR,
          payload: error,
        });
      });
}

export function updateTrip(data) {
  const request = axios.post('/api/tripUpdate', data);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: UPDATE_TRIP,
        payload: response.data,
      });
    });
}

export function deleteTrip(id) {
  const request = axios.delete(`/api/tripDelete?id=${id}`);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: DELETE_TRIP,
        payload: response.data,
      });
    });
}

export function clearTrip() {
  return {
    type: CLEAR_TRIP,
    payload: {
      trip: null,
      updatedTrip: false,
      postDeleted: false,
    },
  };
}

export function clearTripWithReviewer() {
  return {
    type: CLEAR_TRIP_W_REVIEWER,
    payload: {
      trip: null,
      reviewer: null,
    },
  };
}

export function loginUser({ email, password }) {
  const request = axios.post('/api/login', { email, password });

  return dispatch =>
    request.then(response => {
      dispatch({
        type: USER_LOGIN,
        payload: response.data,
      });
    });
}

export function resetPassword({ email }) {
  const request = axios.post('/api/resetPassword', { email });

  return dispatch =>
    request.then(response => {
      dispatch({
        type: RESET_PASSWORD,
        payload: response.data,
      });
    });
}

export function auth() {
  const request = axios.get('/api/auth');

  return dispatch =>
    request.then(response => {
      dispatch({
        type: USER_AUTH,
        payload: response.data,
      });
    });
}

export function addTrip(trip) {
  const request = axios.post('/api/trip', trip);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: ADD_TRIP,
        payload: response.data,
      });
    });
}

export function clearNewTrip() {
  return {
    type: CLEAR_NEW_TRIP,
    payload: null,
  };
}

export function getUsers() {
  const request = axios.get('/api/users');

  return dispatch =>
    request.then(response => {
      dispatch({
        type: GET_USERS,
        payload: response.data,
      });
    });
}

export function userRegister(user) {
  const request = axios.post('/api/register', user);

  return dispatch =>
    request.then(({ data }) => {
      dispatch({
        type: USER_REGISTER,
        payload: data,
      });
    });
}
