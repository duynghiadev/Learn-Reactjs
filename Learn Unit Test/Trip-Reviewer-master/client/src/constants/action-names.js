export const GET_TRIPS = 'get-trips';
export const GET_USERS = 'get-users';
export const ADD_TRIP = 'add-trip';
export const UPDATE_TRIP = 'update-trip';
export const UPDATE_USER = 'update-user';
export const DELETE_TRIP = 'delete-trip';
export const GET_TRIP = 'get-trip';
export const GET_TRIP_ERROR = 'get-trip-error';
export const GET_TRIP_W_REVIEWER = 'get-trip-w-reviewer';
export const GET_TRIP_W_REVIEWER_ERROR = 'get-trip-w-reviewer-error';
export const CLEAR_TRIP_W_REVIEWER = 'clear-trip-w-reviewer';
export const CLEAR_NEW_TRIP = 'clear-new-trip';
export const CLEAR_TRIP = 'clear-trip';
export const USER_REGISTER = 'user-register';
export const USER_LOGIN = 'user-login';
export const RESET_PASSWORD = 'reset-password';
export const USER_AUTH = 'user-auth';
export const GET_USER_REVIEWS = 'get-user-reviews';
