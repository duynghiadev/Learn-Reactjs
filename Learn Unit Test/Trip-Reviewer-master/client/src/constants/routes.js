export const HOME = '/';
export const LOGIN = '/login';
export const RESET = '/reset-password';
export const LOGOUT = '/user/logout';
export const REGISTER = '/user/register';
export const USER_PROFILE = '/user';
export const USER_REVIEWS = '/user/user-reviews';
export const EDIT_PROFILE = '/user/edit-profile';
export const EDIT_POST = '/user/edit-post';
export const ADD_TRIP = '/trip/add';
export const TRIPS = '/trips';
