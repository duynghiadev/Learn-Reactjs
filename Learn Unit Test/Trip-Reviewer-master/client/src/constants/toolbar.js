// prettier-ignore
const toolbar = {
  options: ['image', 'emoji', 'link', 'inline', 'list', 'textAlign', 'remove', 'history'],
  inline: {
    inDropdown: false,
    className: undefined,
    component: undefined,
    dropdownClassName: undefined,
    options: ['bold', 'italic', 'underline', 'strikethrough'],
  },
  list: {
    inDropdown: false,
    className: undefined,
    component: undefined,
    dropdownClassName: undefined,
    options: ['unordered', 'ordered', 'indent', 'outdent'],
  },
  textAlign: {
    inDropdown: false,
    className: undefined,
    component: undefined,
    dropdownClassName: undefined,
    options: ['left', 'center', 'right', 'justify'],
  },
  link: {
    inDropdown: false,
    className: undefined,
    component: undefined,
    popupClassName: undefined,
    dropdownClassName: undefined,
    showOpenOptionOnHover: true,
    defaultTargetOption: '_self',
  },
  emoji: {
    className: undefined,
    component: undefined,
    popupClassName: undefined,
    emojis: [
      '😀', '😁', '😂', '🤣', '😃', '😄', '😅', '😆', '😉', '😊', '😋', '😎', '😍', '😘', '😗', '😙',
      '😚', '🙂', '🤗', '🤔', '😐', '😑', '😶', '🙄', '😏', '😣', '😥', '😮', '🤐', '😯', '😪', '😫',
      '😴', '😌', '😛', '😜', '😝', '🤤', '😒', '😓', '😔', '😕', '🙃', '🤑', '😲', '☹️', '🙁', '😖',
      '😞', '😟', '😤', '😢', '😭', '😦', '😧', '😨', '😩', '😬', '😰', '😱', '😳', '😵', '😡', '😠',
      '😷', '🤒', '🤕', '🤢', '🤧', '😇', '🤠', '🤡', '🤥', '🤓', '😈', '👿', '💀', '👻', '👽', '🤖',
      '💩',

      '👐', '🙌', '👏', '🤝', '👍', '👎', '👊', '✊', '🤛', '🤜', '🤞', '🤘', '👌', '👈', '👉', '👆',
      '👇', '✋', '🤚', '🖐', '🖖', '👋', '🤙', '💪', '🖕', '🙏', '💍', '💄', '💋', '👄', '👅', '👂',
      '👃', '👣', '👁', '👀',

      '👶', '👧', '👦', '👩', '👨', '👵', '👴', '👲', '👳‍♀️', '👳‍♂️', '👱‍♂️', '👱‍♀️', '👮‍♀️', '👮‍♂️', '👷‍♀️',
      '👷‍♂️', '💂‍♀️', '💂‍♂️', '🕵️‍♀️', '🕵️‍♂️', '👩‍⚕️', '👨‍⚕️', '👩‍🌾', '👨‍🌾', '👩‍🍳', '👨‍🍳', '👩‍🎓', '👨‍🎓', '👩‍🎤', '👨‍🎤',
      '👩‍🏫', '👨‍🏫', '👩‍🏭', '👨‍🏭', '👩‍💻', '👨‍💻', '👩‍💼', '👨‍💼', '👩‍🔧', '👨‍🔧', '👩‍🔬', '👨‍🔬', '👩‍🎨', '👨‍🎨', '👩‍🚒', '👨‍🚒',
      '👩‍✈️', '👨‍✈️', '👩‍🚀', '👨‍🚀', '👩‍⚖️', '👨‍⚖️', '👰', '🤵', '👸', '🤴', '🤶', '🎅️', '👼', '🤰', '🙇‍♀️', '🙇‍♂️',
      '💁‍♀️', '💁‍♂️', '🙅‍♀️', '🙅‍♂️', '🙆‍♀️', '🙆‍♂️', '🙋‍♀️', '🙋‍♂️', '🤦‍♀️', '🤦‍♂️', '🤷‍♀️', '🤷‍♂️', '🙎‍♀️', '🙎‍♂️',
      '🙍‍♀️', '🙍‍♂️', '💇‍♀️', '💇‍♂️', '💆‍♀️', '💆‍♂️️', '💅', '🤳', '💃', '🕺', '👯‍♀️', '👯‍♂️', '🕴', '🚶‍♀️', '🚶‍♂️',
      '🏃‍♀️', '🏃‍♂️', '👫', '👭', '👬', '💑', '👩‍❤️‍👩', '👨‍❤️‍👨', '💏', '👩‍❤️‍💋‍👩', '👨‍❤️‍💋‍👨', '👪', '👨‍👩‍👧', '👨‍👩‍👧‍👦', '👨‍👩‍👦‍👦', '👨‍👩‍👧‍👧',
      '👩‍👩‍👦', '👩‍👩‍👧', '👩‍👩‍👧‍👦', '👩‍👩‍👦‍👦', '👩‍👩‍👧‍👧', '👨‍👨‍👦', '👨‍👨‍👧', '👨‍👨‍👧‍👦', '👨‍👨‍👦‍👦', '👨‍👨‍👧‍👧', '👩‍👦', '👩‍👧', '👩‍👧‍👦', '👩‍👦‍👦', '👩‍👧‍👧', '👨‍👦',
      '👨‍👧', '👨‍👧‍👦', '👨‍👦‍👦', '👨‍👧‍👧',

      '🚗', '🚕', '🚙', '🚌', '🚎', '🏎', '🚓', '🚑', '🚒', '🚐', '🚚', '🚛', '🚜', '🛴', '🚲', '🛵',
      '🏍', '🚨', '🚔', '🚍', '🚘', '🚖', '🚡', '🚠', '🚟', '🚃', '🚋', '🚞', '🚝', '🚄', '🚅', '🚈',
      '🚂', '🚆', '🚇', '🚊', '🚉️', '🛫', '🛬', '🛩', '💺', '🛰', '🚀', '🚁', '🛶', '⛵️', '🚤', '🛥',
      '🛳', '⛴', '🚢', '⚓️', '⛽️', '🚧', '🚦', '🚥', '🚏', '🗺', '🗿', '🗽', '🗼', '🏰', '🏯', '🏟',
      '🎡', '🎢', '🎠', '⛲️', '⛱', '🏖', '🏝', '🏜', '🌋', '⛰', '🏔', '🗻', '🏕', '⛺️', '🏠', '🏡',
      '🏘', '🏚', '🏗', '🏭', '🏢', '🏬', '🏣', '🏤', '🏥', '🏦', '🏨', '🏪', '🏫', '🏩', '💒', '🏛',
      '⛪️', '🕌', '🕍', '🕋', '⛩', '🛤', '🛣', '🗾', '🎑', '🏞', '🌅', '🌄', '🌠', '🎇', '🎆', '🌇',
      '🌆', '🏙', '🌃', '🌌', '🌉', '🌁',

      '🍏', '🍎', '🍐', '🍊', '🍋', '🍌', '🍉', '🍇', '🍓', '🍈', '🍒', '🍑', '🍍', '🥝', '🍅', '🍆',
      '🥑', '🥒', '🌶', '🌽', '🥕', '🥔', '🍠', '🥐', '🍞', '🥖', '🧀', '🥚', '🍳', '🥞', '🥓', '🍗',
      '🍖', '🌭', '🍔', '🍟', '🍕', '🥙', '🌮', '🌯', '🥗', '🥘', '🍝', '🍜', '🍲', '🍛', '🍣', '🍱',
      '🍤', '🍚', '🍘', '🍥', '🍢', '🍡', '🍧', '🍨', '🍦', '🍰', '🎂', '🍮', '🍭', '🍬', '🍫', '🍿',
      '🍩', '🍪', '🌰', '🥜', '🍯', '🥛', '🍼', '🍵', '🍶', '🍺', '🍻', '🥂', '🍷', '🥃', '🍸', '🍹',
      '🍾', '🥄', '🍴', '🍽',

      '🐶', '🐱', '🐭', '🐹', '🐰', '🦊', '🐻', '🐼', '🐨', '🐯', '🦁', '🐮', '🐷', '🐽', '🐸', '🐵',
      '🙈', '🙉', '🙊', '🐒', '🐔', '🐧', '🐦', '🐤', '🐣', '🐥', '🦆', '🦅', '🦉', '🦇', '🐺', '🐗',
      '🐴', '🦄', '🐝', '🐛', '🦋', '🐌', '🐚', '🐞', '🐜', '🕷', '🕸', '🦂', '🐢', '🐍', '🦎', '🐙',
      '🦑', '🦐', '🦀', '🐡', '🐠', '🐟', '🐬', '🐳', '🐋', '🦈', '🐊', '🐅', '🐆', '🦍', '🐘', '🦏',
      '🐪', '🐫', '🐃', '🐂', '🐄', '🐎', '🐖', '🐏', '🐑', '🐐', '🦌', '🐕', '🐩', '🐈', '🐓', '🦃',
      '🕊', '🐇', '🐁', '🐀', '🐿', '🐾', '🐉', '🐲', '🌵', '🎄', '🌲', '🌳', '🌴', '🌱', '🌿', '🍀',
      '🎍', '🎋', '🍃', '🍂', '🍁', '🍄', '🌾', '💐', '🌷', '🌹', '🥀', '🌺', '🌸', '🌼', '🌻', '🌞',
      '🌝', '🌛', '🌜', '🌚', '🌕', '🌖', '🌗', '🌘', '🌑', '🌒', '🌓', '🌔', '🌙', '🌎', '🌍', '🌏',
      '💫', '⭐️', '🌟', '✨', '💥', '🔥', '🌪', '🌈', '🌤', '⛅️', '🌥', '🌦', '🌧', '⛈', '🌩', '🌨',
      '⛄️', '💨', '💧', '💦', '🌊',

      '⚽️', '🏀', '🏈', '⚾️', '🏐', '🏉', '🎾', '🎱', '🏓', '🏸', '🥅', '🏒', '🏑', '🏏', '⛳️', '🏹',
      '🎣', '🥊', '🥋', '🎽', '⛸', '🎿', '⛷', '🏂', '🏋️‍♀️', '🏋️‍♂️', '🤼‍♀️', '🤼‍♂️', '🤸‍♀️', '🤸‍♂️', '⛹️‍♀️',
      '⛹️‍♂️', '🤺', '🤾‍♀️', '🤾‍♂️', '🏌️‍♀️', '🏌️‍♂️', '🏇', '🏄‍♀️', '🏄‍♂️', '🏊‍♀️', '🏊‍♂️', '🤽‍♀️', '🤽‍♂️', '🚣‍♀️',
      '🚣‍♂️', '🚵‍♀️', '🚵‍♂️', '🚴‍♀️', '🚴‍♂️', '🏆', '🥇', '🥈', '🥉', '🏅', '🎖', '🏵', '🎗', '🎫', '🎟',
      '🎪', '🤹‍♀️', '🤹‍♂️', '🎭', '🎨', '🎬', '🎤', '🎧', '🎼', '🎹', '🥁', '🎷', '🎺', '🎸', '🎻', '🎲',
      '♟', '🎯', '🎳', '🎮', '🎰',

      '⌚️', '📱', '📲', '💻', '🖥', '🖨', '🖱', '🖲', '🕹', '🗜', '💽', '💾', '💿', '📀', '📼', '📷',
      '📸', '📹', '🎥', '📽', '🎞', '📞', '☎️', '📟', '📠', '📺', '📻', '🎙', '🎚', '🎛', '⏱', '⏲',
      '⏰', '🕰', '⌛️', '⏳', '📡', '🔋', '🔌', '💡', '🔦', '🕯', '🗑', '🛢', '💸', '💵', '💴', '💶',
      '💷', '💰', '💳', '💎', '🔧', '🔨', '🛠', '⛏', '🔩', '⚙️', '⛓', '🔫', '💣', '🔪', '🗡', '🛡',
      '🚬', '🏺', '🔮', '📿', '💈️', '🔭', '🔬', '🕳', '💊', '💉', '🌡', '🚽', '🚰', '🚿', '🛁', '🛀',
      '🛎', '🔑', '🗝', '🚪', '🛋', '🛏', '🛌', '🖼', '🛍', '🛒', '🎁', '🎈', '🎏', '🎀', '🎊', '🎉',
      '🎎', '🏮', '🎐', '📩', '📨', '📧', '💌', '📥', '📤', '📦', '🏷', '📪', '📫', '📬', '📭', '📮',
      '📯', '📜', '📃', '📄', '📑', '📊', '📈', '📉', '🗒', '🗓', '📆', '📅', '📇', '🗃', '🗳', '🗄',
      '📋', '📁', '📂', '🗂', '🗞', '📰', '📓', '📔', '📒', '📕', '📗', '📘', '📙', '📚', '📖', '🔖',
      '🔗', '📎', '🖇', '📐', '📏', '📌', '📍', '🖊', '🖋', '🖌', '🖍', '📝', '🔍', '🔎', '🔏', '🔐',
      '🔒', '🔓',

      '❤️', '💛', '💚', '💙', '💜', '🖤', '💔️', '💕', '💞', '💓', '💗', '💖', '💘', '💝', '💟', '☮️',
      '✝️', '☪️', '🕉', '☸️', '✡️', '🔯', '🕎', '☯️', '☦️', '🛐', '⛎', '♈️', '♉️', '♊️', '♋️', '♌️',
      '♍️', '♎️', '♏️', '♐️', '♑️', '♒️', '♓️', '🆔', '⚛️', '🉑', '☢️', '☣️', '📴', '📳', '🈶', '🈚️',
      '🈸', '🈺', '🈷️', '✴️', '🆚', '💮', '🉐', '㊙️', '㊗️', '🈴', '🈵', '🈹', '🈲', '🅰️', '🅱️', '🆎',
      '🆑', '🅾️', '🆘', '❌', '⭕️', '🛑', '⛔️', '📛', '🚫', '💯', '💢', '♨️', '🚷', '🚯', '🚳', '🚱',
      '🔞', '📵', '🚭', '❗️', '❕', '❓', '❔', '‼️', '⁉️', '🔅', '🔆', '〽️', '⚠️', '🚸', '🔱', '⚜️',
      '🔰', '♻️', '✅', '🈯️', '💹', '❇️', '✳️', '❎', '🌐', '💠', 'Ⓜ️', '🌀', '💤', '🏧', '🚾', '♿️',
      '🅿️', '🈳', '🈂️', '🛂', '🛃', '🛄', '🛅', '🚹', '🚺', '🚼', '🚻', '🚮', '🎦', '📶', '🈁', '🔣',
      'ℹ️', '🔤', '🔡', '🔠', '🆖', '🆗', '🆙', '🆒', '🆕', '🆓', '0️⃣', '1️⃣', '2️⃣', '3️⃣', '4️⃣', '5️⃣',
      '6️⃣', '7️⃣', '8️⃣', '9️⃣', '🔟', '🔢', '#️⃣', '*️⃣', '▶️', '⏸', '⏯', '⏹', '⏺', '⏭', '⏮', '⏩',
      '⏪', '⏫', '⏬', '◀️', '🔼', '🔽', '➡️', '⬅️', '⬆️', '⬇️', '↗️', '↘️', '↙️', '↖️', '↕️', '↔️',
      '↪️', '↩️', '⤴️', '⤵️', '🔀', '🔁', '🔂', '🔄', '🔃', '🎵', '🎶', '➕', '➖', '➗', '✖️', '♾',
      '💲', '💱', '™️', '©️', '®️', '〰️', '➰', '➿', '🔚', '🔙', '🔛', '🔝', '🔜', '✔️', '☑️', '🔘',
      '⚪️', '⚫️', '🔴', '🔵', '🔺', '🔻', '🔸', '🔹', '🔶', '🔷', '🔈', '🔇', '🔉', '🔊', '🔔', '🔕',
      '📣', '📢', '👁‍🗨', '💬', '💭', '🗯', '♠️', '♣️', '♥️', '♦️', '🃏', '🎴', '🀄️', '🕐', '🕑', '🕒',
      '🕓', '🕔', '🕕', '🕖', '🕗', '🕘', '🕙', '🕚', '🕛', '🕜', '🕝', '🕞', '🕟', '🕠', '🕡', '🕢',
      '🕣', '🕤', '🕥', '🕦', '🕧',

      '🏳️', '🏴', '🏁', '🚩', '🏳️‍🌈', '🇦🇫', '🇦🇽', '🇦🇱', '🇩🇿', '🇦🇸', '🇦🇩', '🇦🇴', '🇦🇮', '🇦🇶', '🇦🇬', '🇦🇷',
      '🇦🇲', '🇦🇼', '🇦🇺', '🇦🇹', '🇦🇿', '🇧🇸', '🇧🇭', '🇧🇩', '🇧🇧', '🇧🇾', '🇧🇪', '🇧🇿', '🇧🇯', '🇧🇲', '🇧🇹', '🇧🇴',
      '🇧🇦', '🇧🇼', '🇧🇷', '🇮🇴', '🇻🇬', '🇧🇳', '🇧🇬', '🇧🇫', '🇧🇮', '🇰🇭', '🇨🇲', '🇨🇦', '🇮🇨', '🇨🇻', '🇧🇶', '🇰🇾',
      '🇨🇫', '🇹🇩', '🇨🇱', '🇨🇳', '🇨🇽', '🇨🇨', '🇨🇴', '🇰🇲', '🇨🇬', '🇨🇩', '🇨🇰', '🇨🇷', '🇨🇮', '🇭🇷', '🇨🇺', '🇨🇼',
      '🇨🇾', '🇨🇿', '🇩🇰', '🇩🇯', '🇩🇲', '🇩🇴', '🇪🇨', '🇪🇬', '🇸🇻', '🇬🇶', '🇪🇷', '🇪🇪', '🇪🇹', '🇪🇺', '🇫🇰', '🇫🇴',
      '🇫🇯', '🇫🇮', '🇫🇷', '🇬🇫', '🇵🇫', '🇹🇫', '🇬🇦', '🇬🇲', '🇬🇪', '🇩🇪', '🇬🇭', '🇬🇮', '🇬🇷', '🇬🇱', '🇬🇩', '🇬🇵',
      '🇬🇺', '🇬🇹', '🇬🇬', '🇬🇳', '🇬🇼', '🇬🇾', '🇭🇹', '🇭🇳', '🇭🇰', '🇭🇺', '🇮🇸', '🇮🇳', '🇮🇩', '🇮🇷', '🇮🇶', '🇮🇪',
      '🇮🇲', '🇮🇱', '🇮🇹', '🇯🇲', '🇯🇵', '🎌', '🇯🇪', '🇯🇴', '🇰🇿', '🇰🇪', '🇰🇮', '🇽🇰', '🇰🇼', '🇰🇬', '🇱🇦', '🇱🇻',
      '🇱🇧', '🇱🇸', '🇱🇷', '🇱🇾', '🇱🇮', '🇱🇹', '🇱🇺', '🇲🇴', '🇲🇰', '🇲🇬', '🇲🇼', '🇲🇾', '🇲🇻', '🇲🇱', '🇲🇹', '🇲🇭',
      '🇲🇶', '🇲🇷', '🇲🇺', '🇾🇹', '🇲🇽', '🇫🇲', '🇲🇩', '🇲🇨', '🇲🇳', '🇲🇪', '🇲🇸', '🇲🇦', '🇲🇿', '🇲🇲', '🇳🇦', '🇳🇷',
      '🇳🇵', '🇳🇱', '🇳🇨', '🇳🇿', '🇳🇮', '🇳🇪', '🇳🇬', '🇳🇺', '🇳🇫', '🇰🇵', '🇲🇵', '🇳🇴', '🇴🇲', '🇵🇰', '🇵🇼', '🇵🇸',
      '🇵🇦', '🇵🇬', '🇵🇾', '🇵🇪', '🇵🇭', '🇵🇳', '🇵🇱', '🇵🇹', '🇵🇷', '🇶🇦', '🇷🇪', '🇷🇴', '🇷🇺', '🇷🇼', '🇼🇸', '🇸🇲',
      '🇸🇦', '🇸🇳', '🇷🇸', '🇸🇨', '🇸🇱', '🇸🇬', '🇸🇽', '🇸🇰', '🇸🇮', '🇬🇸', '🇸🇧', '🇸🇴', '🇿🇦', '🇰🇷', '🇸🇸', '🇪🇸',
      '🇱🇰', '🇧🇱', '🇸🇭', '🇰🇳', '🇱🇨', '🇵🇲', '🇻🇨', '🇸🇩', '🇸🇷', '🇸🇿', '🇸🇪', '🇨🇭', '🇸🇾', '🇹🇼', '🇹🇯', '🇹🇿',
      '🇹🇭', '🇹🇱', '🇹🇬', '🇹🇰', '🇹🇴', '🇹🇹', '🇹🇳', '🇹🇷', '🇹🇲', '🇹🇨', '🇹🇻', '🇻🇮', '🇺🇬', '🇺🇦', '🇦🇪', '🇬🇧',
      '󠁧󠁢🇺🇸', '🇺🇾', '🇺🇿', '🇻🇺', '🇻🇦', '🇻🇪', '🇻🇳', '🇼🇫', '🇪🇭', '🇾🇪', '🇿🇲', '🇿🇼',
    ],
  },
  image: {
    className: undefined,
    component: undefined,
    popupClassName: undefined,
    urlEnabled: true,
    uploadEnabled: true,
    alignmentEnabled: true,
    uploadCallback: undefined,
    previewImage: false,
    inputAccept: 'image/gif,image/jpeg,image/jpg,image/png,image/svg',
    alt: { present: false, mandatory: false },
    defaultSize: {
      height: 'auto',
      width: '100%',
    },
  },
  remove: { className: undefined, component: undefined },
  history: {
    inDropdown: false,
    className: undefined,
    component: undefined,
    dropdownClassName: undefined,
    options: ['undo', 'redo'],
  },
};

export default toolbar;
