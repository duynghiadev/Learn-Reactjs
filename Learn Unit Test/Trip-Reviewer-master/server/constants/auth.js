const USER_SALT_I = 10;

module.exports = {
  USER_SALT_I,
};
