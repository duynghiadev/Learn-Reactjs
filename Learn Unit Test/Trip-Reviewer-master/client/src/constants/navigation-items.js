import * as routes from './routes';

export const home = 'HOME';
export const userProfile = 'USER_PROFILE';
export const signup = 'SIGNUP';
export const login = 'LOGIN';
export const userReviews = 'USER_REVIEWS';
export const addTrip = 'ADD_TRIP';
export const logout = 'LOGOUT';

const items = {
  [home]: {
    id: 1,
    icon: 'home',
    text: 'Home',
    link: routes.HOME,
    restricted: false,
  },
  [userProfile]: {
    id: 2,
    icon: 'user',
    text: 'Profile',
    link: routes.USER_PROFILE,
    restricted: true,
  },
  [signup]: {
    id: 3,
    icon: 'user-plus',
    text: 'Signup',
    link: routes.REGISTER,
    restricted: false,
    exclude: true,
  },
  [login]: {
    id: 4,
    icon: 'sign-in-alt',
    text: 'Login',
    link: routes.LOGIN,
    restricted: false,
    exclude: true,
  },
  [userReviews]: {
    id: 5,
    icon: 'edit',
    text: 'Edit',
    link: routes.USER_REVIEWS,
    restricted: true,
  },
  [addTrip]: {
    id: 6,
    icon: 'file-alt',
    text: 'Add',
    link: routes.ADD_TRIP,
    restricted: true,
  },
  [logout]: {
    id: 7,
    icon: 'sign-out-alt',
    text: 'Logout',
    link: routes.LOGOUT,
    restricted: true,
  },
};

export default items;
