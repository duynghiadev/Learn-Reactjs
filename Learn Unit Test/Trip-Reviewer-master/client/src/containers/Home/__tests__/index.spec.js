import React from 'react';
import configureStore from 'redux-mock-store';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';

import EmptyView from '../../../components/Common/EmptyViewAddTrip';
import HomeContainer from '..';
import { GET_TRIPS } from '../../../constants/action-names';
import { getTrips } from '../../../actions';

jest.mock('./../../../components/TripItem', () => 'TripItem');
jest.mock('./../../../components/Common/EmptyViewAddTrip', () => 'EmptyView');
jest.mock('./../../../actions', () => ({
  getTrips: jest.fn(() => ({
    type: GET_TRIPS,
    payload: [],
  })),
}));

const mockStore = configureStore();

const mockComponent = (initialState = {}, props) => {
  const store = mockStore(initialState);

  return <HomeContainer {...props} store={store} />;
};

describe('<HomeContainer />', () => {
  it('should render component', () => {
    const initialState = {
      trips: {},
    };
    const tree = create(mockComponent(initialState)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with a list of items', () => {
    const initialState = {
      trips: {
        list: [
          {
            _id: '_id',
            title: 'title',
            author: 'author',
            price: 'price',
            duration: 'duration',
            rating: 'rating',
          },
        ],
      },
    };
    const tree = create(mockComponent(initialState)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renderItems should return empty view if the list.lenght is small and initialRender - false', () => {
    const initialState = {
      trips: {},
    };
    const instance = shallow(mockComponent(initialState)).dive().instance();
    instance.state.initialRender = false;

    const result = instance.renderItems();

    expect(result).toEqual(
      <div className="empty-view-wrapper">
        <EmptyView />
      </div>
    );
  });

  it('renderLoadMoreButton should not return null if the showLoadmore is true and initialRender - false', () => {
    const initialState = {
      trips: {},
    };
    const instance = shallow(mockComponent(initialState)).dive().instance();
    instance.state.initialRender = false;

    const result = instance.renderLoadMoreButton();

    expect(result).not.toEqual(null);
  });

  it('renderLoadMoreButton should return null if the showLoadmore is false', () => {
    const initialState = {
      trips: {},
    };
    const instance = shallow(mockComponent(initialState)).dive().instance();
    instance.state.showLoadmore = false;

    const result = instance.renderLoadMoreButton();

    expect(result).toEqual(null);
  });

  it('should dispatch getTrips when loadmore is called', () => {
    const list = [];
    const newItemsToLoad = 3;
    const startingFrom = list.length;
    const order = 'desc';
    const initialState = {
      trips: {
        list,
      },
    };

    const instance = shallow(mockComponent(initialState)).dive().instance();

    instance.loadmore();

    expect(getTrips).toHaveBeenCalledWith(newItemsToLoad, startingFrom, order, list);
  });

  it('should update state if componentWillReceiveProps is called with small newTripsCount', () => {
    const setState = jest.fn();
    const nextProps = {
      trips: { newTripsCount: 1 },
    };
    const initialState = {
      trips: {
        list: [],
      },
    };

    const instance = shallow(mockComponent(initialState)).dive().instance();
    instance.setState = setState;

    instance.componentWillReceiveProps(nextProps);

    expect(setState).toHaveBeenCalledWith({ showLoadmore: false });
  });

  it('should update state if componentWillReceiveProps is called with big newTripsCount', () => {
    const setState = jest.fn();
    const nextProps = {
      trips: { newTripsCount: 5 },
    };
    const initialState = {
      trips: {
        list: [],
      },
    };

    const instance = shallow(mockComponent(initialState)).dive().instance();
    instance.setState = setState;
    instance.state.initialRender = false;

    instance.componentWillReceiveProps(nextProps);

    expect(setState).not.toHaveBeenCalled();
  });
});
