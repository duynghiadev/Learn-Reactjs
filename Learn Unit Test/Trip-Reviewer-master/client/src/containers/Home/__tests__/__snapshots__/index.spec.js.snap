// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeContainer /> should render component 1`] = `
<div
  className="home-container limited-width-shadow"
/>
`;

exports[`<HomeContainer /> should render component with a list of items 1`] = `
<div
  className="home-container limited-width-shadow"
>
  <TripItem
    _id="_id"
    author="author"
    duration="duration"
    price="price"
    rating="rating"
    title="title"
  />
</div>
`;
